travel/views.py

```diff
@@ -193,6 +193,32 @@ def create_booking(request, pk):
         }
     return render(request, 'travel/create_booking.html', context=context)
 
+def create_voting(request,pk): 
+    tour = get_object_or_404(Tour, pk=pk)
+    user = User.objects.get(username=str(request.user))
+    if request.method == 'POST':
+        rate = int(request.POST.getlist('voting')[0])
+        try :
+            vote = tour.voting_set.get(user=user)
+        except: 
+            vote = Voting(tour=tour,user=user,star=rate)
+        else: 
+            vote.star = rate
+        
+        try:
+            vote.save()
+        except:
+            messages.error(request, 'Voting fail')
+        else:
+            sumOfRate = 0
+            count = 0
+            for rate in tour.voting_set.all():
```

Comment:
sử dụng query để get lấy giá trị trung bình luôn em nha, không nên query all rồi đi tính thủ công bằng vòng for